core/src/main/scala/org/ensime/indexer/domain.scala

```diff
@@ -35,46 +35,73 @@ final case class PackageName(path: List[String]) extends FullyQualifiedName {
   def parent = PackageName(path.init)
 }
 
-final case class ClassName(pack: PackageName, name: String) extends FullyQualifiedName with DescriptorType {
+sealed abstract class ClassName extends FullyQualifiedName with DescriptorType {
```

Comment:
I know this is what I originally suggested, but I think the three parameter case class is actually a lot easier to understand :smile: sorry for messing you around!